src/neo4j_genai/types.py

```diff
@@ -241,3 +241,12 @@ class Text2CypherRetrieverModel(BaseModel):
     examples: Optional[list[str]] = None
     result_formatter: Optional[Callable[[neo4j.Record], RetrieverResultItem]] = None
     custom_prompt: Optional[str] = None
+
+    @field_validator("custom_prompt")
```

Comment:
I'd say this validation is now deferred to the prompt template class, so we can remove it from here?

Reply:
I understand the concern, though baving it here looks a bit weird to me. Do you think we could have a validation method in the `PromptTemplate`, that could get called when we instantiate the template?